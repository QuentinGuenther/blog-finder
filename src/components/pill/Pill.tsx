import * as React from 'react';

export interface PillProps {
  label: string;
  clickHandler: (event: React.MouseEvent<HTMLButtonElement>) => void;
}

export function Pill (props: PillProps) {
  return (
    <div>
      <button onClick={props.clickHandler}>X</button>
      <p>{props.label}</p>
    </div>
  );
}
