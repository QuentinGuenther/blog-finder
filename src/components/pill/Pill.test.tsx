import * as React from 'react';
import renderer from 'react-test-renderer';
import { shallow, simulate } from 'enzyme';
import { Pill } from './Pill';

describe('Pill should match snapshot', () => {
  it('Pill should match snapshot', () => {
    const handleClick = jest.fn();

    const tree = renderer
      .create(<Pill clickHandler={handleClick} label='r/webdev' />)
      .toJSON();
    
    expect(tree).toMatchSnapshot(); 
  });
});

describe('Tab should be closable', () => {
  it('clicking the X button should trigger the pill to close', () => {
    const handleClick = jest.fn();
    const wrapper = shallow(<Pill clickHandler={handleClick} label='' />)
    wrapper.find('button').simulate('click');
    expect(handleClick).toBeCalled();
  });
});