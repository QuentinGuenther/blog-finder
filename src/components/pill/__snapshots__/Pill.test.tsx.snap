// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pill should match snapshot Pill should match snapshot 1`] = `
<div>
  <button
    onClick={[MockFunction]}
  >
    X
  </button>
  <p>
    
  </p>
</div>
`;
